### PR TITLE
Closes #2542 - Flag that counts the comments of a task has been added

### LINKED ISSUE #2542: Add comment counter flag to task

### PR DESCRIPTION
**Pull request by MM1277**
https://sonarcloud.io/summary/new_code?id=MM1277_taskana&branch=TSK-2542

---
Release Notes:

```
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → (Closes) #&lt;Issue Number&gt;: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_7.0.0_to_7.0.1_h2.sql
Comment (ryzheboka) [inline common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_7.0.0_to_7.0.1_h2.sql:1]: We might not need to update h2 at all. We should discuss this

File: common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql
Comment (ryzheboka) [inline common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql:1]: Next version is 7.1.0 Can you rename the files and change the TASKANA_SCHEMA_VERSION accordingly?

File: common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+-- this script updates the TASKANA database schema from version 7.0.0 to version 7.0.1.
+
+SET search_path = %schemaName%;
+
+INSERT INTO TASKANA_SCHEMA_VERSION (ID, VERSION, CREATED)
+VALUES (nextval('TASKANA_SCHEMA_VERSION_ID_SEQ'), '7.0.1', CURRENT_TIMESTAMP);
+
+ALTER TABLE TASK
+    ADD COLUMN NUMBER_OF_COMMENTS INT4 DEFAULT 0;
```
Comment (ryzheboka) [inline common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql:9]: Why do you use INT4 as type here? Why is the type here different from the type in taskana-schema-postgres?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java [hunk 2]
```diff
@@ -95,7 +96,7 @@ void should_CreateTaskComment_For_TaskThatAlreadyHasComments() throws Exception
 
   @WithAccessId(user = "user-1-2")
   @Test
-  void should_FailToCreateTaskComment_When_UserHasNoWorkbasketPermission() {
+  void should_FailToCreateTaskComment_When_UserHasNoWorkbasketPermission() throws Exception {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:99]: Why did you change this?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java [hunk 3]
```diff
@@ -138,5 +140,6 @@ void should_FailToCreateTaskComment_When_TaskIdIsNullOrNonExisting() throws Exce
     call = () -> taskService.createTaskComment(newTaskCommentForTaskIdNull);
     e = catchThrowableOfType(call, TaskNotFoundException.class);
     assertThat(e.getTaskId()).isNull();
+    assertThat(taskService.getTask(taskWithComments.getId()).getNumberOfComments()).isEqualTo(2);
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:143]: What do you think about modifying the test name to reflect this assertion?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java [hunk 4]
```diff
@@ -87,6 +87,7 @@ void should_CreateTaskComment_For_TaskThatAlreadyHasComments() throws Exception
         taskService.getTaskComments(taskWithComments.getId());
     assertThat(taskCommentsAfterInsert)
         .containsExactlyInAnyOrder(comment1, comment2, taskCommentToCreate);
+    assertThat(taskService.getTask(taskWithComments.getId()).getNumberOfComments()).isEqualTo(3);
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:90]: What do you think about modifying the test name to reflect this assertion?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:1]: Why is this test marked as new, when there already is a test DeleteTaskCommentAccTest.java?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/update/UpdateTaskCommentAccTest.java
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/update/UpdateTaskCommentAccTest.java:1]: Do we need to check the number of TaskComments after a TaskCommentUpdate?

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java [hunk 5]
```diff
@@ -536,6 +536,15 @@ public interface TaskQuery extends BaseQuery<TaskSummary, TaskQueryColumnName> {
    */
   TaskQuery orderByState(SortDirection sortDirection);
 
+  /**
+   * Add the hasComments flag to the query.
+   *
+   * @param hasComments as Boolean. If null, it won't be integrated into the statement. You have to
+   *     set false.
```
Comment (ryzheboka) [inline lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:543]: Please remove the dot after "false"

File: lib/taskana-core/src/main/java/pro/taskana/task/api/models/Task.java [hunk 6]
```diff
@@ -244,6 +244,13 @@ void setPrimaryObjRef(
    */
   String getClassificationCategory();
 
+  /**
+   * Returns the count of the comments of the current {@linkplain Task}.
```
Comment (ryzheboka) [inline lib/taskana-core/src/main/java/pro/taskana/task/api/models/Task.java:248]: You don't need to link Task here because we are in the Task.java

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java [hunk 7]
```diff
@@ -536,6 +536,15 @@ public interface TaskQuery extends BaseQuery<TaskSummary, TaskQueryColumnName> {
    */
   TaskQuery orderByState(SortDirection sortDirection);
 
+  /**
+   * Add the hasComments flag to the query.
+   *
+   * @param hasComments as Boolean. If null, it won't be integrated into the statement. You have to
```
Comment (ryzheboka) [inline lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:542]: What do you mean with "You have to set false"? It should also work when setting it to true

File: qa/hooks/pre-commit
Comment (ryzheboka) [inline qa/hooks/pre-commit:1]: Please delete this file

File: lib/taskana-core-test/src/test/java/acceptance/task/get/GetTaskAccTest.java [hunk 8]
```diff
@@ -112,6 +112,7 @@ void setup() throws Exception {
             .owner("user-1-1")
             .read(true)
             .transferred(false)
+            .numberOfComments(2)
```
[the remark this answers is not in the document]
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/task/get/GetTaskAccTest.java:115]: That is a good idea.

File: common/taskana-common-data/src/main/resources/sql/sample-data/task.sql
[the remark this answers is not in the document]
Comment (MM1277) [reply common/taskana-common-data/src/main/resources/sql/sample-data/task.sql:1]: okay

File: common/taskana-common/src/main/resources/sql/db2/taskana_schema_update_7.0.0_to_7.0.1_db2.sql
[the remark this answers is not in the document]
Comment (MM1277) [reply common/taskana-common/src/main/resources/sql/db2/taskana_schema_update_7.0.0_to_7.0.1_db2.sql:1]: okay

File: common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql
Comment (MM1277) [reply common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql:1]: okay

File: common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql (hunk 1 above)
Comment (MM1277) [reply common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_7.0.0_to_7.0.1_postgres.sql:9]: Because in DBeaver it said INT4 next to the column name of all INT columns. I will change it to INT.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java (hunk 2 above)
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:99]: Because i tried to check the number of comments in this test as well, but deleted it and forgot to delete "throws Exception" too. I will delete it.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:1]: Because it is using the test-api and therefore in the folder taskana-core-test and not in the folder taskana-core. Should i delete the old one that doesn't use the test-api or modify it and delete this new one?

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java (hunk 5 above)
Comment (MM1277) [reply lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:543]: okay. There is also a dot after "false" in the documentation of the method readEquals. Should i delete it there too?

File: lib/taskana-core/src/main/java/pro/taskana/task/api/models/Task.java (hunk 6 above)
Comment (MM1277) [reply lib/taskana-core/src/main/java/pro/taskana/task/api/models/Task.java:248]: okay

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java (hunk 7 above)
Comment (MM1277) [reply lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:542]: i copied that text from the readEquals method that works the same way as the hasComments method. It means that if it is null it will not be automatically set to false, therefore it needs to be set to false if you want to integrate it into the statement with the value false. It does work if it is set to true. Should i rephrase it for the readEquals method too?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/update/UpdateTaskCommentAccTest.java
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/update/UpdateTaskCommentAccTest.java:1]: Yes

File: qa/hooks/pre-commit
Comment (MM1277) [reply qa/hooks/pre-commit:1]: okay

File: lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskAccTest.java [hunk 9]
```diff
@@ -115,6 +115,14 @@ void should_NotSetAttachmentSummariesToNull_When_CreatingNewTaskWithTaskService(
     assertThat(task.asSummary().getAttachmentSummaries()).isNotNull();
   }
 
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_SetNumberOfCommentsToZero_When_CreatingNewTaskWithTaskService() throws Exception {
```
[the remark this answers is not in the document]
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskAccTest.java:120]: okay

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/update/UpdateTaskCommentAccTest.java
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/update/UpdateTaskCommentAccTest.java:1]: okay

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java (hunk 7 above)
Comment (ryzheboka) [reply lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:542]: Yes, thanks :)

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java [hunk 10]
```diff
@@ -120,11 +123,17 @@ void should_CreateTaskComment_When_CopyingAnotherComment() throws Exception {
...
+
+    // Deleting the comment so that the comments remain the same in different tests inside this
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:128]: Why did you add this? How did it work before?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java [hunk 11]
```diff
@@ -120,11 +123,17 @@ void should_CreateTaskComment_When_CopyingAnotherComment() throws Exception {
...
+    // class
+    taskService.deleteTaskComment(taskCommentToCreate.getId());
   }
 
   @WithAccessId(user = "user-1-1")
   @Test
-  void should_FailToCreateTaskComment_When_TaskIdIsNullOrNonExisting() throws Exception {
+  void should_FailToCreateTaskCommentAndUpdateNumberOfComments_When_TaskIdIsNullOrNonExisting()
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:135]: What do you think about should_FailToCreateTaskCommentAndNotUpdateNumberOfComments_When_TaskIdIsNullOrNonExisting()
instead?

**Review by ryzheboka — CHANGES_REQUESTED** [review]

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java (hunk 10 above)
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:128]: sometimes it threw an error for other tests if i ran it without deleting the comment again.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java (hunk 11 above)
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:135]: yes that would be a better name, i will rename it.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java (hunk 10 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:128]: Ok

**Review by ryzheboka — DISMISSED** [review]

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java (hunk 10 above)
Comment (holgerhagen) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:128]: Typically it is a good practice to use different test data in this case to prevent those dependencies. Would it be much effort to create a seperate set of test data for this test?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+            throw new SystemException(
+                "User "
+                    + ex.getCurrentUserId()
+                    + " not authorized on workbasket "
+                    + ex.getWorkbasketId());
+          }
+        });
+    List<TaskComment> taskCommentsAfterDeletionWithAdmin =
+        taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletionWithAdmin).hasSize(0);
```
Comment (holgerhagen) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:134]: btw: what is the difference between isZero and hasSize(0)?

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+    assertThat(taskService.getTask(task1.getId()).getNumberOfComments()).isOne();
+
+    taskanaEngine.runAsAdmin(
+        () -> {
+          try {
+            taskService.deleteTaskComment(comment1.getId());
+          } catch (TaskCommentNotFoundException ex) {
```
Comment (holgerhagen) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:114]: di we really need all these catches in a test? there should be no exception at all....

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java [hunk 14]
```diff
@@ -136,7 +142,10 @@ void deleteTaskComment(String taskCommentId)
           || taskanaEngine.getEngine().isUserInRole(TaskanaRole.ADMIN)
           || taskanaEngine.getEngine().isUserInRole(TaskanaRole.TASK_ADMIN)) {
 
+        Task task =
+            taskService.getTask(taskCommentMapper.findById(taskCommentId).getTaskId());
```
Comment (holgerhagen) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java:146]: why do we read the task if we only need the task id (which we already have to read the task... :-) )

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java [hunk 15]
```diff
@@ -136,7 +142,10 @@ void deleteTaskComment(String taskCommentId)
...
         taskCommentMapper.delete(taskCommentId);
+        taskMapper.decrementNumberOfComments(task.getId(), Instant.now());
```
Comment (holgerhagen) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java:148]: if taskCommentMapper.delete(taskCommentId) fails we do get an exception and this code is not executed, right?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 16]
```diff
@@ -1388,6 +1388,57 @@ void should_GetAllTasks_For_GettingSecondPageFilteredByPorAttributesSortedByType
       assertThat(response.getBody().getLink(IanaLinkRelations.PREV)).isNotNull();
     }
 
+    @Test
+    void should_GetAllTasksWithComments_When_FilteringByHasCommentsIsSetToTrue() {
+      resetDb(); // required because
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:1393]: see comment aboce regarding test data

**Review by holgerhagen — CHANGES_REQUESTED** [review]
Very good work, @MM1277. Thank you. I've just added some small comments. 

Did you test the database update scripts at least on a real postgres and db2 database? This is important.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java (hunk 10 above)
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java:128]: No, it wouldn't be too much effort. I will do as you suggested.

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java (hunk 14 above)
Comment (MM1277) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java]: True, we do not need this. I will fix it.

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java (hunk 16 above)
Comment (MM1277) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java]: I will fix it.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java (hunk 12 above)
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java]: isZero checks if something is 0, hasSize(0) checks if a list has the size 0. hasSize(0) and isEmpty() does the same thing. I will change it to isEmpty() because it is easier to read.

File: common/taskana-common/src/main/resources/sql/db2/taskana_schema_update_7.0.0_to_7.1.0_db2.sql [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+
+INSERT INTO TASKANA_SCHEMA_VERSION (ID, VERSION, CREATED)
+VALUES (TASKANA_SCHEMA_VERSION_ID_SEQ.NEXTVAL, '7.1.0', CURRENT_TIMESTAMP);
+
+ALTER TABLE TASK
+    ADD COLUMN NUMBER_OF_COMMENTS INT DEFAULT 0;
+
+UPDATE TASK t
+SET NUMBER_OF_COMMENTS = (
+	SELECT COUNT(*)
```
[the remark this answers is not in the document]
Comment (MM1277) [reply common/taskana-common/src/main/resources/sql/db2/taskana_schema_update_7.0.0_to_7.1.0_db2.sql]: I will change the scriptsso they include your suggestions. I have tested the database scripts for db2, h2, and postgres with dbeaver.

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java (hunk 15 above)
Comment (MM1277) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java]: yes, from how i understand it, taskCommentMapper.delete(taskCommentId) throws an exception if the delete operation fails. If that method call throws an exceptinon, taskMapper.decrementNumberOfComments(task.getId(), Instant.now()) is not executed.

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java (hunk 13 above)
Comment (MM1277) [reply lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:114]: okay i will do as you suggested.

File: common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_7.0.0_to_7.1.0_h2.sql [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+ALTER TABLE TASK
+    ADD COLUMN NUMBER_OF_COMMENTS INT DEFAULT 0;
+
+UPDATE TASK t
+SET NUMBER_OF_COMMENTS = (
+	SELECT COUNT(tc.ID)
+	FROM TASK_COMMENT tc
+	WHERE tc.TASK_ID = t.ID
+)
+WHERE EXISTS (
```
Comment (holgerhagen) [inline common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_7.0.0_to_7.1.0_h2.sql:16]: It doesn't matter in the end but why didn't you change this query to right outer join as well?

**Review by holgerhagen — CHANGES_REQUESTED** [review]

File: common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_7.0.0_to_7.1.0_h2.sql (hunk 18 above)
Comment (MM1277) [reply common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_7.0.0_to_7.1.0_h2.sql:16]: because h2 doesn't have the update.. set.. from.. syntax, so it didn't work on h2 using the right outer join

**Review by holgerhagen — APPROVED** [review]
